In [27]:
import pandas as pd
import numpy as np

In [28]:
heart = pd.read_csv(r"C:\Users\Carlos\Documents\Python Projects\Heart Disease Research Part 1\heart_disease.csv")

In [29]:
heart.head()

age     sex  trestbps   chol                cp  exang  fbs  thalach  \
0  63.0    male     145.0  233.0    typical angina    0.0  1.0    150.0   
1  67.0    male     160.0  286.0      asymptomatic    1.0  0.0    108.0   
2  67.0    male     120.0  229.0      asymptomatic    1.0  0.0    129.0   
3  37.0    male     130.0  250.0  non-anginal pain    0.0  0.0    187.0   
4  41.0  female     130.0  204.0   atypical angina    0.0  0.0    172.0   

  heart_disease  
0       absence  
1      presence  
2      presence  
3       absence  
4       absence

Let's split the data into two subsets:

yes_hd, which contains data for patients with heart disease

no_hd, which contains data for patients without heart disease

In [30]:
yes_hd = heart[heart.heart_disease == 'presence']
no_hd = heart[heart.heart_disease == 'absence']

In [31]:
yes_hd

age     sex  trestbps   chol               cp  exang  fbs  thalach  \
1    67.0    male     160.0  286.0     asymptomatic    1.0  0.0    108.0   
2    67.0    male     120.0  229.0     asymptomatic    1.0  0.0    129.0   
6    62.0  female     140.0  268.0     asymptomatic    0.0  0.0    160.0   
8    63.0    male     130.0  254.0     asymptomatic    0.0  0.0    147.0   
9    53.0    male     140.0  203.0     asymptomatic    1.0  1.0    155.0   
..    ...     ...       ...    ...              ...    ...  ...      ...   
297  57.0  female     140.0  241.0     asymptomatic    1.0  0.0    123.0   
298  45.0    male     110.0  264.0   typical angina    0.0  0.0    132.0   
299  68.0    male     144.0  193.0     asymptomatic    0.0  1.0    141.0   
300  57.0    male     130.0  131.0     asymptomatic    1.0  0.0    115.0   
301  57.0  female     130.0  236.0  atypical angina    0.0  0.0    174.0   

    heart_disease  
1        presence  
2        presence  
6        presence  
8        presence  
9        presence  
..            ...  
297      presence  
298      presence  
299      presence  
300      presence  
301      presence  

[139 rows x 9 columns]

In [32]:
no_hd

age     sex  trestbps   chol                cp  exang  fbs  thalach  \
0    63.0    male     145.0  233.0    typical angina    0.0  1.0    150.0   
3    37.0    male     130.0  250.0  non-anginal pain    0.0  0.0    187.0   
4    41.0  female     130.0  204.0   atypical angina    0.0  0.0    172.0   
5    56.0    male     120.0  236.0   atypical angina    0.0  0.0    178.0   
7    57.0  female     120.0  354.0      asymptomatic    1.0  0.0    163.0   
..    ...     ...       ...    ...               ...    ...  ...      ...   
288  56.0    male     130.0  221.0   atypical angina    0.0  0.0    163.0   
289  56.0    male     120.0  240.0   atypical angina    0.0  0.0    169.0   
291  55.0  female     132.0  342.0   atypical angina    0.0  0.0    166.0   
295  41.0    male     120.0  157.0   atypical angina    0.0  0.0    182.0   
302  38.0    male     138.0  175.0  non-anginal pain    0.0  0.0    173.0   

    heart_disease  
0         absence  
3         absence  
4         absence  
5         absence  
7         absence  
..            ...  
288       absence  
289       absence  
291       absence  
295       absence  
302       absence  

[164 rows x 9 columns]

To start, we’ll investigate cholesterol levels for patients with heart disease. 

In [33]:
chol_hd = yes_hd.chol

In [34]:
chol_hd

1      286.0
2      229.0
6      268.0
8      254.0
9      203.0
       ...  
297    241.0
298    264.0
299    193.0
300    131.0
301    236.0
Name: chol, Length: 139, dtype: float64

Calculate the mean cholesterol level for patients who were diagnosed with heart disease and print it out. 

In [35]:
chol_hd_mean = np.mean(chol_hd)

In [36]:
chol_hd_mean

251.4748201438849

Import the function from scipy.stats that you can use to test the following null and alternative hypotheses:

Null: People with heart disease have an average cholesterol level equal to 240 mg/dl

Alternative: People with heart disease have an average cholesterol level that is greater than 240 mg/dl

In [37]:
from scipy.stats import ttest_1samp

In [38]:
tstate, pval = ttest_1samp(chol_hd, 240)

In [39]:
pval/2

0.0035411033905155707

Repeat the steps aboce in order to run the same hypothesis test, but for patients in the sample who were not diagnosed with heart disease. 

In [40]:
chol_no_hd = no_hd.chol

In [41]:
chol_no_hd

0      233.0
3      250.0
4      204.0
5      236.0
7      354.0
       ...  
288    221.0
289    240.0
291    342.0
295    157.0
302    175.0
Name: chol, Length: 164, dtype: float64

In [42]:
chol_no_hd_mean = np.mean(chol_no_hd)

In [43]:
chol_no_hd_mean

242.640243902439

In [44]:
tstate, pval = ttest_1samp(chol_no_hd, 240)

In [45]:
pval/2

0.26397120232220506

How many patients are there in this dataset? 

In [48]:
num_patients = len(heart)

In [49]:
num_patients

303

Calculate the number of patients with fasting blood sugar greater than 120.

In [54]:
num_highfbs_patients = sum(heart.fbs)

In [55]:
num_highfbs_patients

45.0

By some estimates, about 8% of the U.S. population had diabetes (diagnosed or undiagnosed) in 1988 when this data was collected.

If this sample were representative of the population, approximately how many people would you expect to have diabetes?

In [56]:
0.08 * num_patients

24.240000000000002

Run binomial test

In [57]:
from scipy.stats import binom_test
pval = binom_test(num_highfbs_patients, num_patients, .08, alternative='greater')

In [58]:
pval

4.689471951448875e-05